File: c/concepts/loops.c
#include <stdio.h>
#include "concepts.h"

// Loops
void demonstrateLoops() {
    printf("\n=== Loops ===\n");

    // For loop
    printf("Counting with for loop:\n");
    for (int i = 1; i < 5; ++i) {
        printf("%d ", i);
    }
    printf("\n");

    // While loop
    printf("Counting with while loop:\n");
    int count = 1;
    while (count < 5) {
        printf("%d ", count);
        count++;
    }
    printf("\n");
}

File: perl/concepts/loops.pl
# perl/concepts/loops.pl
use strict;
use warnings;

sub demonstrate_loops {
    print "\n=== Loops ===\n";

    # C-style for loop. The variable must be declared with `my`.
    print "Counting with for loop:\n";
    for (my $i = 1; $i < 5; $i++) {
        print "$i ";
    }
    print "\n";

    # While loop
    print "Counting with while loop:\n";
    my $count = 1;
    while ($count < 5) {
        print "$count ";
        $count++;
    }
    print "\n";
}

1;
